### PR TITLE
Use max size constant from C++ library

### PR DESCRIPTION
**Pull request by mcobzarenco**
This was a [review comment](https://github.com/reinfer/blingfire-rs/pull/1#discussion_r300887286)

### CODE REVIEW

File: blingfire/src/lib.rs [hunk 1]
```diff
@@ -125,7 +122,7 @@ mod tests {
 
     #[test]
     fn text_to_words_string_too_long() {
-        let source = String::from_utf8(vec![b'.'; MAX_SOURCE_LENGTH + 1]).unwrap();
+        let source = String::from_utf8(vec![b'.'; MAX_TEXT_LENGTH + 1]).unwrap();
         let mut destination = String::new();
         assert!(text_to_words(&source, &mut destination).is_err());
```
Comment (cristicbz) [inline blingfire/src/lib.rs]: also, i know this was pre-existing but could you check that the *actual* error is `SourceTooLong`

File: blingfire/src/lib.rs [hunk 2]
```diff
@@ -22,13 +25,6 @@ pub fn text_to_sentences(source: &str, destination: &mut String) -> Result<()> {
 
 type Tokenizer = unsafe extern "C" fn(*const c_char, c_int, *mut c_char, c_int) -> c_int;
 
-// The maximum length is when the destination length can't fit in the maximum blingfire array. Worst
-// case, the source is tokenized by adding a space after every character + a null character at the
-// end.
-//
-// The maximum array size is defined in blingfire-sys/BlingFire/blingfireclient.library/inc/FALimits.h
-const MAX_SOURCE_LENGTH: usize = 1_000_000_000 / 2 - 2;
```
Comment (cristicbz) [inline blingfire/src/lib.rs:30]: I would keep this as `MAX_TEXT_LENGTH / 2 - 2`. But you can merge it either way, up to you.

**Review by cristicbz — APPROVED** [review]
Only change is the test, otherwise :+1:

File: blingfire/src/lib.rs (hunk 2 above)
Comment (mcobzarenco) [reply blingfire/src/lib.rs:30]: I still don't get why and it seems to rely on specifics of how the tokenizer works

File: blingfire/src/lib.rs (hunk 1 above)
Comment (mcobzarenco) [reply blingfire/src/lib.rs]: done

**Pull request merged**